MAVProxy/modules/mavproxy_link.py

```diff
@@ -453,11 +478,29 @@ def master_msg_handling(self, m, master):
                 self.mpstate.vehicle_name = 'AntennaTracker'
 
         elif mtype == 'STATUSTEXT':
-            if m.text != self.status.last_apm_msg or time.time() > self.status.last_apm_msg_time+2:
-                (fg, bg) = self.colors_for_severity(m.severity)
-                self.mpstate.console.writeln("APM: %s" % mp_util.null_term(m.text), bg=bg, fg=fg)
-                self.status.last_apm_msg = m.text
-                self.status.last_apm_msg_time = time.time()
+            key = "%s.%s" % (m.get_srcSystem(), m.get_srcComponent())
+            if key not in self.status.statustexts_by_sysidcompid:
+                self.status.statustexts_by_sysidcompid[key] = {}
+            if m.id not in self.status.statustexts_by_sysidcompid[key]:
+                self.status.statustexts_by_sysidcompid[key][m.id] = {}
```

Comment:
I'd prefer creating a class for this, not a dictionary

Reply:
Done.